src/modules/esl-anchornav/core/esl-anchornav.ts

```diff
@@ -0,0 +1,224 @@
+import {ExportNs} from '../../esl-utils/environment/export-ns';
+import {ESLBaseElement} from '../../esl-base-element/core';
+import {attr, decorate, listen, memoize, prop, ready} from '../../esl-utils/decorators';
+import {debounce, microtask} from '../../esl-utils/async';
+import {getViewportForEl} from '../../esl-utils/dom/scroll';
+import {ESLEventUtils, ESLIntersectionTarget} from '../../esl-event-listener/core';
+import {ESLAnchor} from './esl-anchor';
+
+import type {DelegatedEvent, ESLIntersectionEvent} from '../../esl-event-listener/core';
+import type {ESLAnchorData, ESLAnchornavRender} from './esl-anchornav-types';
+
+/**
+ * ESLAnchornav
+ * @author Dmytro Shovchko
+ *
+ * ESLAnchornav is a component that collects content anchors from the page and provides anchor navigation
+ */
+@ExportNs('Anchornav')
+export class ESLAnchornav extends ESLBaseElement {
+  public static override is = 'esl-anchornav';
+  public static _renderers: Map<string, ESLAnchornavRender> = new Map();
+
+  /** Gets renderer by name */
+  public static getRenderer(name: string): ESLAnchornavRender | undefined {
+    return this._renderers.get(name);
+  }
+
+  /** Sets renderer */
+  public static setRenderer(renderer: ESLAnchornavRender): void;
+  public static setRenderer(name: string, renderer: ESLAnchornavRender): void;
+  public static setRenderer(name: string | ESLAnchornavRender, renderer?: ESLAnchornavRender): void {
+    if (typeof name !== 'string') return this.setRenderer('default', name);
+    if (typeof name === 'string' && renderer) this._renderers.set(name, renderer);
+  }
+
+  @prop('esl:anchornav:activechanged') public ACTIVECHANGED_EVENT: string;
+  @prop('esl:anchornav:updated') public UPDATED_EVENT: string;
+  @prop('[esl-anchor]') protected ANCHOR_SELECTOR: string;
+
+  /** Item renderer which is used to build inner markup */
+  @attr({defaultValue: 'default', name: 'renderer'}) public rendererName: string;
+
+  protected _active: ESLAnchorData;
+  protected _anchors: ESLAnchorData[] = [];
+  protected _items: Map<string, Element> = new Map();
+  protected _offset: number;
+
+  /** Active anchor */
+  public get active(): ESLAnchorData {
+    return this._active;
+  }
+  public set active(value: ESLAnchorData) {
+    if (this._active === value) return;
+    this._active = value;
+    this._onActiveChange(value);
+  }
+
+  /** Anchors list */
+  protected get $anchors(): HTMLElement[] {
+    return this._anchors.map(({$anchor}) => $anchor);
+  }
+
+  /** Anchornav offset */
+  public get offset(): number {
+    return this._offset || 0;
+  }
+  public set offset(value: number) {
+    if (this._offset === value) return;
+    this._offset = value;
+    memoize.clear(this, '$viewport');
+    this.$$on(this._onAnchorIntersection);
+  }
+
+  /** Anchornav item renderer */
+  protected get itemRenderer(): ESLAnchornavRender | undefined {
+    return ESLAnchornav.getRenderer(this.rendererName) || ESLAnchornav.getRenderer('default');
+  }
```

Comment:
Fallback already provided on the attribute level
```suggestion
  protected get itemRenderer(): ESLAnchornavRender | undefined {
    return ESLAnchornav.getRenderer(this.rendererName);
  }
```